src/main/java/org/folio/entlinks/service/authority/AuthoritySourceFileService.java

```diff
@@ -83,12 +84,14 @@ public AuthoritySourceFile update(@Nonnull UUID id, AuthoritySourceFile modified
 
   public void deleteById(UUID id) {
     log.debug("deleteById:: Attempt to delete AuthoritySourceFile by [id: {}]", id);
-
-    if (!repository.existsById(id)) {
-      throw new AuthoritySourceFileNotFoundException(id);
+    var authoritySourceFile = repository.findById(id)
+        .orElseThrow(() -> new AuthoritySourceFileNotFoundException(id));
+    if (!authoritySourceFile.getType().equals(FOLIO.getValue())) {
```

Comment:
I don't think it's good idea to check entity values with dto's enum. If we don't have enum for entity, I suggest to create it and use it there and also in the entity field.. We have only to possible values: folio and local.

if type in entity is not required we could get an NPE in this condition check.

Reply:
And it should be `source`, not `type`